docs/deployment-guide-terraform.md

```diff
@@ -0,0 +1,424 @@
+  type        = string
+  default     = "usgovernment"
+}
+
+variable "metadata_host" {
+  description = "The metadata host for the Azure Cloud e.g. management.azure.com"
+  type        = string
+  default     = "management.usgovcloudapi.net"
+}
+
+variable "location" {
+  description = "The Azure region for most Mission LZ resources"
+  type        = string
+  default     = "usgovvirginia"
+}
+```
+
+```terraform
+provider "azurerm" {
+  features {}
+  
+  environment     = var.environment # e.g. 'public' or 'usgovernment'
+  metadata_host   = var.metadata_host # e.g. 'management.azure.com' or 'management.usgovcloudapi.net'
+}
+```
+
+For the supported `environment` values, see this doc: <https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#environment/>
+
+For the supported `metadata_host` values, see this doc: <https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#metadata_host/>
+
+For more endpoint mappings between AzureCloud and AzureUsGovernment: <https://docs.microsoft.com/en-us/azure/azure-government/compare-azure-government-global-azure#guidance-for-developers/>
+
+### Terraform Providers
+
+The [development container definition](../.devcontainer/Dockerfile) downloads the required Terraform plugin providers during the container build so that the container can be transported to an air-gapped network.
+
+The container sets the `TF_PLUGIN_CACHE_DIR` environment variable, which Terraform uses as the search location for locally installed providers.
+
+If you are not using the [development container](../.devcontainer) to deploy or if the `TF_PLUGIN_CACHE_DIR` environment variable is not set, Terraform will automatically attempt to download the provider from the internet when you execute the `terraform init` command.
+
+See the development container [README](/.devcontainer/README.md) for more details on building and running the container.
+
+### Terraform Backends
+
+The default templates write a state file directly to disk locally to where you are executing terraform from. If you wish to change the output directory you can set the path directly in the terraform backend block located in the main.tf file via the path variable in the backend configuration block.
+
+```terraform
+terraform {
+  backend "local" {
+    path = "relative/path/to/terraform.tfstate"
+  }
+
+  required_version = ">= 1.0.3"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "= 2.71.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "= 3.1.0"
+    }
+    time = {
+      source  = "hashicorp/time"
+      version = "0.7.2"
+    }
+  }
+}
+```
+
+To find more information about setting the backend see [Local Backend](https://www.terraform.io/docs/language/settings/backends/local.html),  if you wish to AzureRM backend please see [AzureRM Backend](https://www.terraform.io/docs/language/settings/backends/azurerm.html)
+
+## Cleanup
```

Comment:
Should we document `destroy`ing in the reverse order of `apply`?

e.g. 

1. apply mlz/main.tf
2. apply tier3/main.tf
3. destroy tier3/main.tf
4. destroy mlz/main.tf

Reply:
I added a statement and an example for destroying in reverse order of creating. 🥇